test_apps/smokeTests/gridTools.cpp

```diff
@@ -0,0 +1,594 @@
+        count++;
+    }
+
+    size_t expectedCount = 1;
+    for ( auto dim : dims )
+        expectedCount *= dim;
+
+    time = Wasp::GetTime() - t0;
+
+    return expectedCount;
+}
+
+size_t TestConstCoordItr( 
+    const Grid *g,
+    size_t& count, 
+    size_t& disagreements,
+    double& time 
+) {
+    Grid::ConstCoordItr itr;
+    Grid::ConstCoordItr enditr = g->ConstCoordEnd();
+
+    double t0 = Wasp::GetTime();
+    itr = g->ConstCoordBegin();
+    
+    std::vector< size_t > dims = g->GetDimensions();
+
+    for ( ; itr!=enditr; ++itr) {
+        size_t i = count % dims[X];
+        size_t j = ( count / dims[X] ) % dims[Y];
+        size_t k = count / ( dims[X] * dims[Y] );
+        size_t ijk[] = { i, j, k };
+        double coords[3];
+
+        bool disagree = false;
+        g->GetUserCoordinates( ijk, coords );
+        for ( size_t dim=0; dim<dims.size(); dim++ ) {
+            if ( isNotEqual( (*itr)[dim], coords[dim] ) ) {
+                disagree = true;
+            }
+        }
+        if ( disagree ) {
+            disagreements++;
+        }
+
+        count++;
+    }
+
+    size_t expectedCount = 1;
+    for ( auto dim : dims )
+        expectedCount *= dim;
+
+    time = Wasp::GetTime() - t0;
+
+    return expectedCount;
+}
+
+void PrintStats( double rms, size_t numMissingValues, size_t disagreements ) {
+    cout << "    RMS error:                                     " << rms << endl;
+    cout << "    Missing value count:                           " << numMissingValues << endl;
+    cout << "    AccessIJK() vs GetValue() disagreement count:  " << disagreements << endl;
+    cout << endl;
+}
+
+bool RunTest(
+    Grid* grid
+) {
+    bool rc = 0;
+    double rms;
+    size_t numMissingValues;
+    size_t disagreements;
+
+    if ( CompareIndexToCoords( grid, rms, numMissingValues, disagreements ) ) {
+        cout << "       *** Error reported in " << grid->GetType() << " grid***" << endl;
+        rc = 1;
+    }
+
+    if ( grid->GetInterpolationOrder() == 0 )
+        cout << "  nearestNeighbor: " << endl;
+    else
+        cout << "  linear:          " << endl;
+    PrintStats( rms, numMissingValues, disagreements );
+
+    return rc;
+}
+
+bool RunTests( 
+    Grid* grid, 
+    const std::vector<std::string> &tests, 
+    float minVal, 
+    float maxVal 
+) {
+
+    std::vector< size_t > dims = grid->GetDimensions();
+    size_t x = dims[X];
+    size_t y = dims[Y];
+    size_t z = 1;
+    if ( dims.size() == 3 ) 
+        z = dims[Z];
+
+    int rc=0;
+    std::string type = grid->GetType();
+    
+    cout << "=======================================================" << endl << endl;
+    if ( std::find( tests.begin(), tests.end(), "Constant" ) != tests.end() ) {
+        cout << type << " " << x << "x" << y << "x" << z << " Constant field:" << endl;
+        MakeConstantField( grid, maxVal );
+       
+        grid->SetInterpolationOrder( linear );
+        if ( RunTest( grid ) != 0 )
+            rc = 1;
```

Comment:
Return true on success, false on failure per our error conventions and common practice.

Reply:
Fixed.